navigationrail/navigationrail/src/main/java/com/microsoft/device/dualscreen/navigationrail/NavigationRailView.kt

```diff
@@ -0,0 +1,522 @@
+    ) {
+        if (skipAnimation || !useAnimation || Build.VERSION.SDK_INT <= Build.VERSION_CODES.LOLLIPOP) {
+            child.translationY = childTop.toFloat()
+        } else {
+            child.animate()
+                .setInterpolator(animationInterpolator)
+                .translationY(childTop.toFloat())
+        }
+    }
+
+    /**
+     * Returns the margin of the [NavigationRailMenuView] from the top of it's parent.
+     * This is useful when the [NavigationRailView] contains a header view.
+     */
+    private fun calculateMenuTopMargin(): Int {
+        val topMargin =
+            resources.getDimensionPixelSize(com.google.android.material.R.dimen.mtrl_navigation_rail_margin)
+        return headerView?.let {
+            it.bottom + topMargin
+        } ?: kotlin.run { 0 }
+    }
+
+    private fun getGravity() = menuGravity and Gravity.VERTICAL_GRAVITY_MASK
+
+    private fun getChildMenu(): NavigationRailMenuView? {
+        for (i in 0..childCount) {
+            val child = getChildAt(i)
+            if (child is NavigationRailMenuView) {
+                return child
+            }
+        }
+        return null
+    }
+
+    /**
+     * Checks if the newly requested positioning for the buttons matches the existing one.
+     */
+    private fun NavigationRailMenuView.doesChildCountMatch(
+        startBtnCount: Int,
+        endBtnCount: Int
+    ): Boolean {
+        return startBtnCount >= 0 && endBtnCount >= 0 && childCount == startBtnCount + endBtnCount
+    }
+
+    /**
+     * Skip the animation on the first layout pass.
+     */
+    private fun shouldSkipAnimation(menu: NavigationRailMenuView): Boolean {
+        for (i in 0 until menu.childCount) {
+            if (menu.getChildAt(i).left != 0) {
+                return true
+            }
+        }
+        return false
+    }
+
+    /**
+     * Synchronize the [startBtnCount] and [endBtnCount].
+     */
+    private fun syncBtnCount(startBtnCount: Int, endBtnCount: Int) {
+        this.topBtnCount = startBtnCount
+        this.bottomBtnCount = endBtnCount
+    }
+
+    override fun onInterceptTouchEvent(ev: MotionEvent): Boolean {
+        if (!allowFlingGesture) {
+            return super.onInterceptTouchEvent(ev)
+        }
+
+        onSwipeListener.onTouchEvent(ev)
+
+        if (onSwipeListener.onInterceptTouchEvent(ev)) {
+            return true
+        }
+        return super.onInterceptTouchEvent(ev)
+    }
+
+    @SuppressLint("ClickableViewAccessibility")
+    override fun onTouchEvent(ev: MotionEvent?): Boolean {
+        if (!allowFlingGesture) {
+            return super.onTouchEvent(ev)
+        }
+        return onSwipeListener.onTouchEvent(ev)
+    }
+}
+
+fun View.absY(): Int {
+    val location = IntArray(2)
+    this.getLocationOnScreen(location)
+    return location[1]
+}
+
+private fun NavigationRailMenuView.colorButtons() {
+    for (i in 0 until childCount) {
+        getChildAt(i).background = ColorDrawable(Color.CYAN)
```

Comment:
All the time is Color.CYAN?

Reply:
The method was used for debugging purposes. 
I forgot to delete it.